afkak/client.py

```diff
@@ -800,48 +801,64 @@ def _next_id(self):
         self.correlation_id = (self.correlation_id + 1) % 2**31
         return self.correlation_id
 
-    def _make_request_to_broker(self, broker, requestId, request, **kwArgs):
-        """Send a request to the specified broker."""
-        def _timeout_request(broker, requestId):
-            """The time we allotted for the request expired, cancel it."""
-            try:
-                # FIXME: This should be done by calling .cancel() on the Deferred
-                # returned by the broker client.
-                broker.cancelRequest(requestId, reason=RequestTimedOutError(
-                    'Request: {} cancelled due to timeout'.format(requestId)))
-            except KeyError:  # pragma: no cover This should never happen...
-                log.exception('ERROR: Failed to find key for timed-out '
-                              'request. Broker: %r Req: %d',
-                              broker, requestId)
-                raise
+    def _make_request_to_broker(self, broker, correlationId, request, expectResponse=True, min_timeout=None):
+        """
+        Send a request to the specified broker, with timeout logic.
+        """
+        rr = _ReprRequest(request)
+        issued = self.reactor.seconds()
+        cell = [None]  # single-element list: nonlocal for Python 2
+
+        if min_timeout is not None:
+            timeout = max(self.timeout, min_timeout)
+        else:
+            timeout = self.timeout
+
+        def _mrtb_timeout():
```

Comment:
It's not clear to me what "mrtb" means?

Reply:
It is an abbreviation for "make request to broker," i.e., this method's name.

I plan to clean up some of this logging stuff in a later PR, probably after merging #37 to avoid conflicts.

Reply:
Yeah, sorry about that :-)